### ISSUE TITLE
Despite gmp is installed, nextcloud bookmarks wont install because gmp library missing

### ISSUE THREAD

**Issue by devplan44** [nextcloud/bookmarks#935]
**Describe the bug**
I installed now my Nextcloud on a VPS server. Everything installed ok without errors.
When attempting to install bookmarks app it complains of missing gmp library.

**To Reproduce**
1. Apps
2. Click on "Organization"
3. Click on "Bookmarks" "Download and Enable"
4. See error: App "Bookmarks" cannot be installed because the following dependencies are not fulfilled: The library gmp is not available.

**Expected behavior**
Should ask for my password and install

**Desktop (please complete the following information):**
 - OS: CentOS 7
 - Browser Chrome
 - Version 80.0.3987.122 (Official Build) (64-bit)

**Server (please complete the following information):**
 - OS:  Ubuntu 18.04.4 LTS (GNU/Linux 4.15.0-88-generic x86_64)
 - HTTP server: Apache/2.4.29 (Ubuntu)
 - Database MySQL 5.7.29
 - PHP version:  v7.4
 - Nextcloud version: 16.0.8
 - Bookmarks app version: doesn't say
 - Activated Nextcloud Apps: $ sudo -u www-data php occ app:list` doesn't work
but I have Carnet and Music app installed

**Insert your webserver log here**
[Sat Mar 07 22:39:21.271450 2020] [core:notice] [pid 774] AH00094: Command line: '/usr/sbin/apache2'
[Sat Mar 07 23:19:47.052227 2020] [mpm_prefork:notice] [pid 774] AH00169: caught SIGTERM, shutting down
[Sat Mar 07 23:20:07.497495 2020] [mpm_prefork:notice] [pid 761] AH00163: Apache/2.4.29 (Ubuntu) configured -- resuming normal operations
[Sat Mar 07 23:20:07.523761 2020] [core:notice] [pid 761] AH00094: Command line: '/usr/sbin/apache2'
[Sat Mar 07 23:20:08.222769 2020] [php7:error] [pid 773] [client 149.224.228.151:43860] PHP Fatal error:  Uncaught Doctrine\\DBAL\\DBALException: Failed to connect to the database: An exception occurred in driver: SQLSTATE[HY000] [2002] No such file or directory in /var/www/nextcloud/lib/private/DB/Connection.php:64\nStack trace:\n#0 /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(448): OC\\DB\\Connection->connect()\n#1 /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(410): Doctrine\\DBAL\\Connection->getDatabasePlatformVersion()\n#2 /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(354): Doctrine\\DBAL\\Connection->detectDatabasePlatform()\n#3 /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(710): Doctrine\\DBAL\\Connection->getDatabasePlatform()\n#4 /var/www/nextcloud/lib/private/DB/Connection.php(151): Doctrine\\DBAL\\Connection->setTransactionIsolation(2)\n#5 /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DriverManager.php(181): OC\\DB\\Connection->__construct(Array, Object(Doctrine\\DBAL\\Driver\\PDOMySql\\Driver), Ob in /var/www/nextcloud/lib/private/DB/Connection.php on line 64
[Sat Mar 07 23:20:08.223346 2020] [php7:error] [pid 773] [client 149.224.228.151:43860] PHP Fatal error:  Uncaught Doctrine\\DBAL\\DBALException: Failed to connect to the database: An exception occurred in driver: SQLSTATE[HY000] [2002] No such file or directory in /var/www/nextcloud/lib/private/DB/Connection.php:64\nStack trace:\n#0 /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(448): OC\\DB\\Connection->connect()\n#1 /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(410): Doctrine\\DBAL\\Connection->getDatabasePlatformVersion()\n#2 /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(354): Doctrine\\DBAL\\Connection->detectDatabasePlatform()\n#3 /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(710): Doctrine\\DBAL\\Connection->getDatabasePlatform()\n#4 /var/www/nextcloud/lib/private/DB/Connection.php(151): Doctrine\\DBAL\\Connection->setTransactionIsolation(2)\n#5 /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DriverManager.php(181): OC\\DB\\Connection->__construct(Array, Object(Doctrine\\DBAL\\Driver\\PDOMySql\\Driver), Ob in /var/www/nextcloud/lib/private/DB/Connection.php on line 64
[Sat Mar 07 23:43:11.616843 2020] [mpm_prefork:notice] [pid 761] AH00169: caught SIGTERM, shutting down
[Sat Mar 07 23:43:33.227150 2020] [mpm_prefork:notice] [pid 779] AH00163: Apache/2.4.29 (Ubuntu) configured -- resuming normal operations
[Sat Mar 07 23:43:33.236419 2020] [core:notice] [pid 779] AH00094: Command line: '/usr/sbin/apache2'
[Sat Mar 07 23:46:57.359937 2020] [authz_core:error] [pid 950] [client 149.224.228.151:32926] AH01630: client denied by server configuration: /var/www/nextcloud/data/.ocdata
[Sun Mar 08 00:20:05.721881 2020] [mpm_prefork:notice] [pid 779] AH00169: caught SIGTERM, shutting down
[Sun Mar 08 00:20:27.220088 2020] [mpm_prefork:notice] [pid 761] AH00163: Apache/2.4.29 (Ubuntu) configured -- resuming normal operations
[Sun Mar 08 00:20:27.221532 2020] [core:notice] [pid 761] AH00094: Command line: '/usr/sbin/apache2'
[Sun Mar 08 00:52:54.356989 2020] [mpm_prefork:notice] [pid 761] AH00171: Graceful restart requested, doing restart
[Sun Mar 08 00:52:54.864613 2020] [mpm_prefork:notice] [pid 761] AH00163: Apache/2.4.29 (Ubuntu) configured -- resuming normal operations
[Sun Mar 08 00:52:54.864661 2020] [core:notice] [pid 761] AH00094: Command line: '/usr/sbin/apache2'
[Sun Mar 08 00:53:10.342170 2020] [mpm_prefork:notice] [pid 761] AH00169: caught SIGTERM, shutting down
[Sun Mar 08 00:53:10.566325 2020] [mpm_prefork:notice] [pid 1916] AH00163: Apache/2.4.29 (Ubuntu) configured -- resuming normal operations
[Sun Mar 08 00:53:10.566399 2020] [core:notice] [pid 1916] AH00094: Command line: '/usr/sbin/apache2'
[Sun Mar 08 00:55:18.570296 2020] [mpm_prefork:notice] [pid 1916] AH00169: caught SIGTERM, shutting down
[Sun Mar 08 00:55:40.258232 2020] [mpm_prefork:notice] [pid 802] AH00163: Apache/2.4.29 (Ubuntu) configured -- resuming normal operations
[Sun Mar 08 00:55:40.260389 2020] [core:notice] [pid 802] AH00094: Command line: '/usr/sbin/apache2'
[Sun Mar 08 01:26:52.152171 2020] [mpm_prefork:notice] [pid 802] AH00169: caught SIGTERM, shutting down
[Sun Mar 08 01:27:12.211698 2020] [mpm_prefork:notice] [pid 833] AH00163: Apache/2.4.29 (Ubuntu) configured -- resuming normal operations
[Sun Mar 08 01:27:12.212440 2020] [core:notice] [pid 833] AH00094: Command line: '/usr/sbin/apache2'
[Sun Mar 08 01:37:02.255995 2020] [mpm_prefork:notice] [pid 833] AH00171: Graceful restart requested, doing restart
[Sun Mar 08 01:37:02.686516 2020] [mpm_prefork:notice] [pid 833] AH00163: Apache/2.4.29 (Ubuntu) configured -- resuming normal operations
[Sun Mar 08 01:37:02.686545 2020] [core:notice] [pid 833] AH00094: Command line: '/usr/sbin/apache2'
[Sun Mar 08 01:37:16.609575 2020] [mpm_prefork:notice] [pid 833] AH00169: caught SIGTERM, shutting down
[Sun Mar 08 01:37:16.817609 2020] [mpm_prefork:notice] [pid 1262] AH00163: Apache/2.4.29 (Ubuntu) configured -- resuming normal operations
[Sun Mar 08 01:37:16.817695 2020] [core:notice] [pid 1262] AH00094: Command line: '/usr/sbin/apache2'
[Sun Mar 08 01:42:24.169774 2020] [mpm_prefork:notice] [pid 1262] AH00171: Graceful restart requested, doing restart
[Sun Mar 08 01:42:24.401902 2020] [mpm_prefork:notice] [pid 1262] AH00163: Apache/2.4.29 (Ubuntu) configured -- resuming normal operations
[Sun Mar 08 01:42:24.401930 2020] [core:notice] [pid 1262] AH00094: Command line: '/usr/sbin/apache2'
[Sun Mar 08 01:42:29.415373 2020] [mpm_prefork:notice] [pid 1262] AH00169: caught SIGTERM, shutting down
[Sun Mar 08 01:42:29.661988 2020] [mpm_prefork:notice] [pid 1481] AH00163: Apache/2.4.29 (Ubuntu) configured -- resuming normal operations
[Sun Mar 08 01:42:29.662079 2020] [core:notice] [pid 1481] AH00094: Command line: '/usr/sbin/apache2'
[Sun Mar 08 01:54:20.681358 2020] [mpm_prefork:notice] [pid 1481] AH00169: caught SIGTERM, shutting down
[Sun Mar 08 01:54:43.104858 2020] [mpm_prefork:notice] [pid 769] AH00163: Apache/2.4.29 (Ubuntu) configured -- resuming normal operations
[Sun Mar 08 01:54:43.107807 2020] [core:notice] [pid 769] AH00094: Command line: '/usr/sbin/apache2'
[Sun Mar 08 02:24:11.997534 2020] [authz_core:error] [pid 775] [client 149.233.177.253:33352] AH01630: client denied by server configuration: /var/www/nextcloud/data/.ocdata

**Nextcloud log (nextcloud/data/nextcloud.log)**
{"reqId":"Cgu8hOTgRJkBPZ7ziMGk","level":3,"time":"2020-03-08T01:39:09+00:00","remoteAddr":"149.233.177.253","user":"mgarbe","app":"$

**Insert your Nextcloud log here**
none

**Here is some gmp related data:**
----------------------------------------------------------
**# php -i**
phpinfo()
PHP Version => 7.4.3

System => Linux 87649007540349 4.15.0-88-generic #88-Ubuntu SMP Tue Feb 11 20:11:34 UTC 2020 x86_64
Build Date => Feb 23 2020 07:24:28
Server API => Command Line Interface
Virtual Directory Support => disabled
Configuration File (php.ini) Path => /etc/php/7.4/cli
Loaded Configuration File => /etc/php/7.4/cli/php.ini
Scan this dir for additional .ini files => /etc/php/7.4/cli/conf.d
Additional .ini files parsed => /etc/php/7.4/cli/conf.d/10-mysqlnd.ini,
/etc/php/7.4/cli/conf.d/10-opcache.ini,
/etc/php/7.4/cli/conf.d/10-pdo.ini,
/etc/php/7.4/cli/conf.d/15-xml.ini,
/etc/php/7.4/cli/conf.d/20-bz2.ini,
/etc/php/7.4/cli/conf.d/20-calendar.ini,
/etc/php/7.4/cli/conf.d/20-ctype.ini,
/etc/php/7.4/cli/conf.d/20-curl.ini,
/etc/php/7.4/cli/conf.d/20-dom.ini,
/etc/php/7.4/cli/conf.d/20-exif.ini,
/etc/php/7.4/cli/conf.d/20-ffi.ini,
/etc/php/7.4/cli/conf.d/20-fileinfo.ini,
/etc/php/7.4/cli/conf.d/20-ftp.ini,
/etc/php/7.4/cli/conf.d/20-gd.ini,
/etc/php/7.4/cli/conf.d/20-gettext.ini,
/etc/php/7.4/cli/conf.d/20-gmp.ini,
/etc/php/7.4/cli/conf.d/20-iconv.ini,
/etc/php/7.4/cli/conf.d/20-imagick.ini,
/etc/php/7.4/cli/conf.d/20-intl.ini,
/etc/php/7.4/cli/conf.d/20-json.ini,
/etc/php/7.4/cli/conf.d/20-mbstring.ini,
/etc/php/7.4/cli/conf.d/20-mysqli.ini,
/etc/php/7.4/cli/conf.d/20-pdo_mysql.ini,
/etc/php/7.4/cli/conf.d/20-phar.ini,
/etc/php/7.4/cli/conf.d/20-posix.ini,
/etc/php/7.4/cli/conf.d/20-readline.ini,
/etc/php/7.4/cli/conf.d/20-shmop.ini,
/etc/php/7.4/cli/conf.d/20-simplexml.ini,
/etc/php/7.4/cli/conf.d/20-sockets.ini,
/etc/php/7.4/cli/conf.d/20-sysvmsg.ini,
/etc/php/7.4/cli/conf.d/20-sysvsem.ini,
/etc/php/7.4/cli/conf.d/20-sysvshm.ini,
/etc/php/7.4/cli/conf.d/20-tokenizer.ini,
/etc/php/7.4/cli/conf.d/20-xmlreader.ini,
/etc/php/7.4/cli/conf.d/20-xmlwriter.ini,
/etc/php/7.4/cli/conf.d/20-xsl.ini,
/etc/php/7.4/cli/conf.d/20-zip.ini

PHP API => 20190902
PHP Extension => 20190902
Zend Extension => 320190902
Zend Extension Build => API320190902,NTS
PHP Extension Build => API20190902,NTS
Debug Build => no
Thread Safety => disabled
Zend Signal Handling => enabled
Zend Memory Manager => enabled
Zend Multibyte Support => provided by mbstring
IPv6 Support => enabled
DTrace Support => available, disabled

Registered PHP Streams => https, ftps, compress.zlib, php, file, glob, data, http, ftp, compress.bzip2, phar, zip
Registered Stream Socket Transports => tcp, udp, unix, udg, ssl, tls, tlsv1.0, tlsv1.1, tlsv1.2, tlsv1.3
Registered Stream Filters => zlib.*, string.rot13, string.toupper, string.tolower, string.strip_tags, convert.*, consumed, dechunk, bzip2.*, convert.iconv.*

This program makes use of the Zend Scripting Language Engine:
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.3, Copyright (c), by Zend Technologies

 _______________________________________________________________________

Configuration

bz2

BZip2 Support => Enabled
Stream Wrapper support => compress.bzip2://
Stream Filter support => bzip2.decompress, bzip2.compress
BZip2 Version => 1.0.6, 6-Sept-2010

calendar

Calendar support => enabled

Core

PHP Version => 7.4.3

.
.
gmp

gmp support => enabled
GMP version => 6.1.2
.
.
-------------------------------------------------
**# php -m**
[PHP Modules]
bz2
calendar
Core
ctype
curl
date
dom
exif
FFI
fileinfo
filter
ftp
gd
gettext
gmp
hash
iconv
imagick
intl
json
libxml
mbstring
mysqli
mysqlnd
openssl
pcntl
pcre
PDO
pdo_mysql
Phar
posix
readline
Reflection
session
shmop
SimpleXML
sockets
sodium
SPL
standard
sysvmsg
sysvsem
sysvshm
tokenizer
xml
xmlreader
xmlwriter
xsl
Zend OPcache
zip
zlib

[Zend Modules]
Zend OPcache
--------------------------------------------------

/etc/php/7.4/cli/conf.d/20-gmp.ini:
-----------------------------------
; configuration for php gmp module
; priority=20
extension=gmp.so

**Comment by marcelklehr** [comment]
Mmmh, qfter a quick glance I see you're using nc v16, which doesn't support php v7.4, afaik

**Comment by devplan44** [comment]
Thanks for your hint.

Updated now to v18 but bookmarks install still complained about the same thing.
Then I reckoned, I might have two php versions installed and along perhaps the wrong gmp version.
I became suspicious when recently seeing in myphpadmin the reported php version being 7.2

So, I installed the php7.2-gmp and vola, the bookmark app installed.

Now my question is: should I keep both php versions? Does Bookmarks require php 7.2?
When I query for php version:
**~# php -v**
PHP 7.4.3 (cli) (built: Feb 23 2020 07:24:28) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.3, Copyright (c), by Zend Technologies
